### PR TITLE
completed week 7 assesmment

### PR DESCRIPTION
**Pull request by tammytran96**
# Submitting Week 7 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X] Coding Challenges
- [X] Coding Concepts
- [X] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,34 +13,109 @@
...
 
 // b) Create the function that makes the test pass.
+// pseudocode
+// create a function called descriptiveSentence
+// input is an array of objects
+// output is an array with a sentence about each person with their name capitalized
+// iterate over the array of objects with .map() to be able to access each object
+// while in the .map() function, use dot notation to access the values of the names and occupations inside the object
+// we won't need to modify the occupation but we need to capitalize the name
+    // split the array with a space in between so that it doesn't split by letters, but by words - so that we have an array of arrays of name's values (a nested array) - make this into a variable so that it's not confusing
```
Comment (elysemontano) [inline code-challenges.test.js:37]: Excellent work!  Great break down of the problem 👏

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,34 +13,109 @@
...
+// output is the sum of all the numbers cubed
+// iterate over the array using .map() and cube each number - set that to a variable
+// declare another variable that starts at 0 - this will be the start of the running sum
+// iterate over the mapped cubed array using a for loop and add each value of the array to the 0 variable so we can keep a running count
+// return the total of the cubed numbers
+
+const cubedSum = (array) => {
+  cubedArray = array.map(value => value **3)
+  let runningSum = 0
+  for(let i = 0; i < cubedArray.length; i++){
```
Comment (elysemontano) [inline code-challenges.test.js:115]: Awesome work!  A possible refactor to consider would be to use a higher order function like reduce that will allow you to not only access the current value, but will also allow you to have a running sum in it as well.  Definitely a higher order function to dig into at some point.

File: rails-commenting.rb [hunk 3]
```diff
@@ -6,39 +6,39 @@
 
 # FILE: app/controller/blog_posts_controller.rb
 
-# ---1)
+# ---1) Below is a class called BlogPostsController that contains multiple methods within it. It is taking in the same methods and instances that the superclass ApplicationController has.
```
Comment (elysemontano) [inline rails-commenting.rb:9]: This section looks excellent!  Great job 🤩

**Review by elysemontano — APPROVED** [review]
Outstanding work on this last assessment Tammy 🎉!!  Your code looks great, you do an excellent job breaking down the problems in your pseudocode, your interview answers were amazing and thorough, and the rails section was top notch!  I want to encourage you to try doing refactors when possible and think about edge cases and how you can write tests for some edge cases.
